src/main/java/com/ethanyidong/bunny/Bunny.java

```diff
@@ -0,0 +1,81 @@
+package com.ethanyidong.bunny;
+
+import com.ethanyidong.bunny.arg.InvalidCommandException;
+import com.ethanyidong.bunny.cmd.ExecutableCommand;
+import com.ethanyidong.bunny.cmd.TokenizedCommand;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Scanner;
+
+public class Bunny {
+    public static final String DISABLE_SAVE = "nosave";
+    public static void main(String[] args) {
+        final boolean saveEnabled = args.length < 1 || !args[0].equals(DISABLE_SAVE);
+        Scanner in = new Scanner(System.in);
+        BunnySession bunny = new BunnySession();
+
+        String home = System.getProperty("user.home");
+        Path saveFilePath = Paths.get(home, "bunny.aof");
+        boolean saveFileExists = Files.exists(saveFilePath);
+
+        if(saveFileExists && saveEnabled) {
+            bunny.setIsSuppressed(true);
+            try {
+                BufferedReader reader = Files.newBufferedReader(saveFilePath, Charset.forName("UTF-8"));
+                String saveFileLine = reader.readLine();
+                while(saveFileLine != null) {
+                    runCommandString(bunny, saveFileLine);
+                    saveFileLine = reader.readLine();
+                }
+                reader.close();
+            } catch (Exception _ex){
+                System.out.println("Error reading save file! Continuing...");
+            } finally {
+                bunny.setIsSuppressed(false);
+            }
+        }
+
+        bunny.printMessage("Hello! I'm Bunny.\nWhat can I do for you?");
```

Comment:
Avoid the usage of magic literals. You can refactor them to static final string constants. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)